modules/irmaseal-pkg-server.nix

```diff
@@ -0,0 +1,92 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.irmaseal-pkg;
+in
+{
+
+  options.services.irmaseal-pkg = {
+
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        This option enables the IRMAseal-PKG server.
+      '';
+    };
+
+    package = mkOption {
+      type = types.path;
+      default = pkgs.irmaseal-pkg;
+      description = "The IRMAseal-PKG package to use";
+    };
+
+    host = mkOption {
+      type = types.str;
+      default = "127.0.0.1";
+      description = ''
+        Host to bind this service to
+      '';
+    };
+
+    irma = mkOption {
+      type = types.str;
+      default = "https://irma-noauth.demo.sarif.nl";
+      description = ''
+        URL of the IRMA go server to use for authentication
+      '';
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 8087;
+      description = ''
+        TCP port to bind this service to
+      '';
+    };
+
+    publicKeyPath = mkOption {
+      type = types.path;
+      default = ./pkg.pub;
+      description = ''
+        Path to the public key
+      '';
+    };
+
+    secretKeyPath = mkOption {
+      type = types.path;
+      default = ./pkg.sec;
+      description = ''
+        Path to the private key
+      '';
+    };
+
+  };
+
+  config = mkIf cfg.enable {
+
+    environment.systemPackages = [ cfg.package ];
+
+    systemd.services.irmaseal-pkg = {
+      description = "IRMAseal PKG HTTP server";
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig = {
+        ExecStart = ''
+          ${cfg.package}/bin/irmaseal-pkg server \
+            --host '${cfg.host}' \
+            --irma '${cfg.irma}' \
+            --port ${toString cfg.port} \
+            --public '${cfg.publicKeyPath}' \
+            --secret '${cfg.secretKeyPath}'
+        '';
+        Restart = "always";
+        DynamicUser = true;
+        PrivateTmp = true;
+        ReadOnlyPaths = "${cfg.publicKeyPath} ${cfg.secretKeyPath}";
```

Comment:
second thing that I wasn't sure about. Since those keys are generated, it can't be readonlypath, there is another option called ReadWritePaths, that should fit our case.

Reply:
Hmm, the flow is usually like this: the user generates the keys, and then passes the keys to the service. It is not actually written by the service. Would that be okay?